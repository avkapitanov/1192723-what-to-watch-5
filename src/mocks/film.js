export const promoFilm = {
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "id": 1,
  "is_favorite": true,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const filmWIthChangedStatus = {
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "id": 1,
  "is_favorite": false,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const film = {
  id: 1,
  title: `We need to talk about Kevin`,
  genre: [`Drama`],
  year: 2011,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
  background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/We_need_to_talk_about_Kevin.jpg`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  backgroundColor: `#E1DFDE`,
  rating: 3.8,
  scoresCount: 123240,
  description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
  director: `Lynne Ramsay`,
  starring: [
    `Tilda Swinton`,
    `John C. Reilly`,
    `Ezra Miller`
  ],
  runTime: 112,
  isFavorite: false
};
