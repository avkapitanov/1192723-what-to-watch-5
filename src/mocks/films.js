export const films = [{
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "id": 1,
  "is_favorite": true,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Pulp Fiction`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "background_color": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "rating": 1.5,
  "scores_count": 1635992,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "run_time": 153,
  "genre": `Crime`,
  "released": 1994,
  "id": 2,
  "is_favorite": true,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Midnight Special`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg`,
  "background_color": `#828585`,
  "description": `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
  "rating": 3.3,
  "scores_count": 67815,
  "director": `Jeff Nichols`,
  "starring": [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst `],
  "run_time": 112,
  "genre": `Action`,
  "released": 2016,
  "id": 3,
  "is_favorite": true,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  "name": `Matrix`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "background_color": `#B9B27E`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "rating": 4.4,
  "scores_count": 1503092,
  "director": `Wachowski Brothers`,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "run_time": 136,
  "genre": `Action`,
  "released": 1999,
  "id": 4,
  "is_favorite": true,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Legend`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
  "background_color": `#E1DAD7`,
  "description": `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
  "rating": 3.5,
  "scores_count": 138487,
  "director": `Brian Helgeland`,
  "starring": [`Tom Hardy`, `Emily Browning`, `Taron Egerton`],
  "run_time": 132,
  "genre": `Crime`,
  "released": 2015,
  "id": 5,
  "is_favorite": false,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}];

export const normalizedFilms = {
  "1": {
    id: 1,
    title: `We need to talk about Kevin`,
    genre: [`Drama`],
    year: 2011,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/We_need_to_talk_about_Kevin.jpg`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    backgroundColor: `#E1DFDE`,
    rating: 3.8,
    scoresCount: 123240,
    description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
    director: `Lynne Ramsay`,
    starring: [
      `Tilda Swinton`,
      `John C. Reilly`,
      `Ezra Miller`
    ],
    runTime: 112,
    isFavorite: false
  },
  "2": {
    id: 2,
    title: `What We Do in the Shadows`,
    genre: [`Comedy`, `Drama`],
    year: 2019,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/What-We-Do-in-the-Shadows.jpg`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/What-We-Do-in-the-Shadows.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/What-We-Do-in-the-Shadows.jpg`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `#A39E81`,
    rating: 7.2,
    scoresCount: 6173,
    description: `A look into the daily (or rather, nightly) lives of three vampires who've lived together for over 100 years, in Staten Island.`,
    director: `Jemaine Clement`,
    starring: [
      `Kayvan Novak`,
      `Matt Berry`,
      `Natasia Demetriou`
    ],
    runTime: 30,
    isFavorite: false
  },
  "3": {
    id: 3,
    title: `Bronson`,
    genre: [`Action`],
    year: 2008,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `#73B39A`,
    rating: 3.6,
    scoresCount: 109661,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    starring: [
      `Tom Hardy`,
      `Kelly Adams`,
      `Luing Andrews`
    ],
    runTime: 92,
    isFavorite: false
  }
};
