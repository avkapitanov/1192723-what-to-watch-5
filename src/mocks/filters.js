export const filterGenres = [
  `All genres`,
  `Drama`,
  `Comedy`,
  `Horror`,
  `Romance`,
  `Sci-Fi`
];

export const selectedFilterGenre = `Drama`;
