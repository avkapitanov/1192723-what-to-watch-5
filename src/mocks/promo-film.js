const promoFilm = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  poster: `./img/the-grand-budapest-hotel-poster.jpg`,
  background: `./img/bg-the-grand-budapest-hotel.jpg`
};

export default promoFilm;
