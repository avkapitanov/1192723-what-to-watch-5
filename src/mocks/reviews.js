import {nanoid} from 'nanoid';
import {getRandomElements, getRandomInteger, generateDate, getRandomFloat} from "../utils";

const REVIEWS_FILM_CARD_COUNT = 9;
const MIN_REVIEW_SENTENCE = 1;
const MAX_REVIEW_SENTENCE = 4;
const MIN_REVIEW_RATING = 6;
const MAX_REVIEW_RATING = 10;

const generateAuthor = () => {
  const authors = [
    `Kate Muir`,
    `Matthew Lickona`,
    `Bill Goodykoontz`,
    `Paula Fleri-Soler`,
    `Amanda Greever`,
    `Cameron Owens`
  ];

  const randomIndex = getRandomInteger(0, authors.length - 1);

  return authors[randomIndex];
};

const generateText = () => {
  const text = [
    `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.`,
    `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    `Fusce tristique felis at fermentum pharetra.`,
    `Aliquam id orci ut lectus varius viverra.`,
    `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.`,
    `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
    `Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
    `Sed sed nisi sed augue convallis suscipit in sed felis.`,
    `Aliquam erat volutpat.`,
    `Nunc fermentum tortor ac porta dapibus.`,
    `In rutrum ac purus sit amet tempus.`
  ];

  const sentenceCount = getRandomInteger(MIN_REVIEW_SENTENCE, MAX_REVIEW_SENTENCE);

  return getRandomElements(text, sentenceCount).join(` `);
};

const generateReview = () => {
  const author = generateAuthor();

  return {
    id: nanoid(),
    author,
    rating: getRandomFloat(MIN_REVIEW_RATING, MAX_REVIEW_RATING),
    date: generateDate(),
    text: generateText()
  };
};

const reviews = new Array(REVIEWS_FILM_CARD_COUNT).fill().map(generateReview);

export default reviews;
