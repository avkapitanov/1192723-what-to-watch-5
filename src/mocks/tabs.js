export const tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];

export const activeTab = `Overview`;
