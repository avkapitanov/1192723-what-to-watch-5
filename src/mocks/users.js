export const authInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
};
