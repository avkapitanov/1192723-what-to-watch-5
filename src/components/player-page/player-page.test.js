import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";

import {normalizedFilms} from "../../mocks/films";
import PlayerPage from "./player-page";

const films = Object.values(normalizedFilms);

const mockStore = configureStore([]);
const store = mockStore({
  DATA: {
    films: {
      entities: films
    },
    filmId: 1
  }
});

it(`PlayerPage component render correctly`, () => {
  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter history={browserHistory}>
          <PlayerPage
            fetchFilm={() => {}}
          />
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});


