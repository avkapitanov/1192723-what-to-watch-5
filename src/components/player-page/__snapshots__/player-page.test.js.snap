// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPage component render correctly 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/What-We-Do-in-the-Shadows.jpg"
    src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={0}
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "NaN%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        What We Do in the Shadows
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
